### DISCUSSION TITLE
Welcome to Weather-Notification Discussions!

### DISCUSSION THREAD

**Discussion by Dhruv-Jain31** [Dhruv-Jain31/Weather-Notification#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Dhruv-Jain31** [comment]
Hi as we know that we have been assigned with a agile project on weather notification. We'll be discussing the plan of it tomorrow or some any day when you're comfortable.

**Comment by Kashishturan** [comment]
Hii, Dhruv. It's nice to collaborate with you . Looking forward to work with you . Yes, sure we will be discussing the ideas about this project from now on.

**Reply by Dhruv-Jain31** [reply]
Looking forward for this collaboration

**Comment by Kashishturan** [comment]
Sure, for now we have to keep in mind the constraints we got i.e, the time limit and the programming language i.e, python. Keeping in mind these to we would start exploring ideas and inform each other over here.

**Comment by Dhruv-Jain31** [comment]
Hi, as you know that today we discussed the overview of our project. Here is the brief summary of the things discussed:
1. We'll be primarily using Html, CSS for our frontend work.
2. For our backend work we'll be using Django since python is the constraint given.
3. We'll be using API requests to fetch up the data.

I hope that you are comfortable with the tech stack. I you feel any sort of problem feel free to reach out.

**Comment by Kashishturan** [comment]
I am not familiar with Html and CSS.

**Comment by Kashishturan** [comment]
In python, i am comfortable so yes, I am comfortable in using Django.

**Comment by Kashishturan** [comment]
And we will discuss the details about code tomorrow and then start working on building the code and for the code our main focus will be on python as it is the main language in which our code has to be built .

**Comment by Kashishturan** [comment]
Hey, dhruv
As we have to show our visual representation idea to sir. What is in your mind regarding that ?

**Comment by Kashishturan** [comment]
I do have an idea. Lets share each other's idea over here !

**Comment by Dhruv-Jain31** [comment]
I also have an idea. Currently I am busy, will be here shortly.

**Comment by Kashishturan** [comment]
Dhruv, are you available now?

**Comment by Kashishturan** [comment]
My idea is that we will have weather notification box in center  and to highlight that we will choose dark color for the box and light background . What you say ?

**Comment by Kashishturan** [comment]
I am attaching picture of one idea of mine and colors' for background and box we will discuss later, for now you see the idea and let me know your opinion.
![Idea1](https://github.com/Dhruv-Jain31/Weather-Notification/assets/142867357/31925e28-872f-4e84-a313-9e1a372f53ff)

**Comment by Dhruv-Jain31** [comment]
Hi, your template is good enough but we have to add other variables as well. I am dropping a rough template of a project that was assigned to some other pair. There project was on contact list.
![visual representation ](https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/7ae89c87-998d-4ab7-8f9e-4b9fc32a44ad)

**Comment by Kashishturan** [comment]
Could you please suggest the other variable required in our topic?

**Comment by Dhruv-Jain31** [comment]
along with the features of our project we have to also include user permissions in our template like we have to define that what user can do, similarly what a admin can do etc. Basically we have to follow KANBAN model of visualization of things.

**Comment by Kashishturan** [comment]
Ok , will think of more ideas . Similarly you also drop visual representation of your idea on our topic.

**Comment by Dhruv-Jain31** [comment]
for our project that we our doing we also have a feature of comparing weather's of two cities. so it gives user a liberty of comparing weather within the same site. She has to not hover on some other website.

**Comment by Dhruv-Jain31** [comment]
[KANBAN BOARD]

**To Do:**
- Create database to store weather data
- Develop API to retrieve weather data
- Design website layout
- Implement website functionality
- Test website and fix bugs

**In Progress:**
- Developing API to retrieve weather data
- Designing website layout

**Done:**
- Created database to store weather data

**Priority:**
- High
- Medium
- Low

**Due Date:**
- 2023-09-30
- 2023-10-07
- 2023-10-14

**Blockers:**
- Waiting for API to be completed
- Waiting for design approval

[SWIMLANES]

**Homepage**
- Create homepage layout
- Implement weather forecast functionality
- Implement city comparison functionality

**About Us**
- Create about us page layout
- Add company information
- Add team member profiles

**Contact Us**
- Create contact us page layout
- Add contact form
- Add contact information

**Team:**
- John Smith
- Jane Doe
- Peter Jones



Here is a example of a KANBAN board. I hope now you have the better clarity of the things. Although it is quite complicated one but we have a complete liberty to alter it according to our requirements. I hope that this one will go fine. We don't need to go for a image template we can easily create this one on notepad or MS word. For visual representation we can use colour-coding. This one is more complicated one our will be way more simpler due to our project so don't go into intricacies of it. For now I am raising a issue in the issue section as well for the kanban board so that we can show something in the issues tab as well. We would adding stuff related to coding part there but for the first time i hope that this issue would be fine.

**Comment by Kashishturan** [comment]
Okay . But do post your visual representation idea which is specifically related to our project and not any other.

**Comment by Kashishturan** [comment]
Hey Dhruv, i got one more idea. I am attaching the pic of my idea here:
![idea2 devops](https://github.com/Dhruv-Jain31/Weather-Notification/assets/142867357/7cdc1b42-99a4-4653-b49e-df8a6dce9133)

**Comment by Dhruv-Jain31** [comment]
i have done with project. Yours one is correct but we to show the use cases and features of our project. That can be visually done by flowchart or basic sequence relationship. I am dropping a image so to have a better clarity of things.
[
![flowchart](https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/5677a7b7-5a78-4b94-801c-3d392abc6f22)
](url) i hope it's clear now

**Reply by Dhruv-Jain31** [reply]
just tap in the area where image is there you'll be able to see the clear image

**Comment by Dhruv-Jain31** [comment]
I am also dropping a basic workflow of our project.
[
![download](https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/7f869d3a-7860-490c-ad59-513549357a1b)
](url)

**Reply by Dhruv-Jain31** [reply]
just tap on the image area again to see a clear picture.

**Comment by Dhruv-Jain31** [comment]
along with it i have also designed a KANBAN board for maintaining the workflow of our project. I have added you as a collaborator so that you can edit the things. It's purely for our reference.
[
[kanban board for project 2.docx](https://github.com/Dhruv-Jain31/Weather-Notification/files/12718473/kanban.board.for.project.2.docx)
](url)

**Comment by Kashishturan** [comment]
Yes cool , so basically we have workflow as well as visual idea of how our outcome may look like .

**Comment by Dhruv-Jain31** [comment]
ya it is just a workflow to make our work streamlined. so that there can clarity of things and we can trace our progress as well.

**Comment by Kashishturan** [comment]
Hey, dhruv since we have started working on code, i feel should start updating it in parts.

**Reply by Dhruv-Jain31** [reply]
yes working on it

**Comment by Kashishturan** [comment]
And also, we have to update on KANBAN board.

**Reply by Dhruv-Jain31** [reply]
okay

**Comment by Dhruv-Jain31** [comment]
I am making necessary files

**Comment by Kashishturan** [comment]
Dhruv though there are some glitches I am working on the code and so do you . Update me here any details.

**Reply by Dhruv-Jain31** [reply]
No problem I'll try to resolve it

**Comment by Kashishturan** [comment]
![p2e1](https://github.com/Dhruv-Jain31/Weather-Notification/assets/142867357/d4262c36-36da-41dc-9233-0a4c6f7f93f5)


See though files are imported and necessary modules downloaded I am getting this glitch. Please you try running it once.

**Reply by Dhruv-Jain31** [reply]
i have resolved this glitch. It running fine on my machine

**Comment by Kashishturan** [comment]
As well I will update this on our KANBAN Board.

**Reply by Dhruv-Jain31** [reply]
ok

**Comment by Dhruv-Jain31** [comment]
I am just facing some issues in fetching the data from api. They might have not activated the api key. Will again try this later

**Comment by Kashishturan** [comment]
Okay. I am also working on Api.

**Comment by Kashishturan** [comment]
Rest code I have committed in code section.

**Reply by Dhruv-Jain31** [reply]
ok

**Comment by Kashishturan** [comment]
![p2e2](https://github.com/Dhruv-Jain31/Weather-Notification/assets/142867357/adc812ae-56db-4d05-9cde-72e796d4bce6)

**Comment by Kashishturan** [comment]
So this is the error we are getting.

**Reply by Dhruv-Jain31** [reply]
yes trying to resolve it

**Comment by Kashishturan** [comment]
![p2e3](https://github.com/Dhruv-Jain31/Weather-Notification/assets/142867357/041c25fd-44fb-45c8-ade8-78158a1012e5)
See output is ready and just API recovery is pending.

**Reply by Dhruv-Jain31** [reply]
yes

**Comment by Kashishturan** [comment]
Hey Dhruv as you wrote a mail to company regarding Api thing . Did you get any reply?

**Reply by Dhruv-Jain31** [reply]
yes they replied. and they activated the api key. there was a glitch from there side and api is working fine enough now

**Reply by Dhruv-Jain31** [reply]
[
<img width="680" alt="call_api" src="https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/c545680e-24c8-42b7-bceb-896196dac862">
](url)
you can clearly see

**Comment by Dhruv-Jain31** [comment]
I have changed the code base. There was a problem with the old code base. I have shared you the video pls have a look and also drop a review as well.

**Comment by Dhruv-Jain31** [comment]
Due to size of the video it can't be uploaded here

**Comment by Kashishturan** [comment]
Yes, output is nice and also new code base is working well.

**Comment by Kashishturan** [comment]
So yes, we will go with the new code base as old when was giving glitch .

**Comment by Kashishturan** [comment]
![output](https://github.com/Dhruv-Jain31/Weather-Notification/assets/142867357/bd996d9b-b8ad-4c88-ab53-54b7fc4e1abd)
See, the code is running well and provided output is also good.

**Comment by Kashishturan** [comment]
Also, I have added our KANBAN board file.

**Reply by Dhruv-Jain31** [reply]
ok

**Comment by Kashishturan** [comment]
As well providing you with the link , if you wish to like to add or edit any part.
https://in.docworkspace.com/d/sIM-uwuNQz_3pqAY

**Reply by Dhruv-Jain31** [reply]
done

**Comment by Dhruv-Jain31** [comment]
[
<img width="514" alt="image" src="https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/3a23d977-66f2-4080-92bc-8c013f3e0b67">
](url)
final kanban board

**Comment by Kashishturan** [comment]
Okayy

**Comment by Kashishturan** [comment]
Dhruv, we are finally done with our project. It was nice working with you . Hope our presentation goes well.

**Reply by Dhruv-Jain31** [reply]
Great job, Kashish! I'm sure our presentation will go great. It was a pleasure working with you on this project, and I'm excited to see all the hard work we put in pay off.

**Comment by Dhruv-Jain31** [comment]
Heyy, I am attaching the final screenshots of feedback collected. It was nice working with you. Excited to see all the hard work that we put in pay off. Great job!

**Comment by Dhruv-Jain31** [comment]
[
<img width="559" alt="Screenshot 2023-10-30 235307" src="https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/e9655c30-4a76-499b-9150-361461299b95">
](url)

[
<img width="557" alt="Screenshot 2023-10-30 235514" src="https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/6bb9c970-a5d6-4365-bafc-c3ff6876d8fe">
](url)

[
<img width="534" alt="Screenshot 2023-10-30 235603" src="https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/a4475095-882c-4a89-a39b-db670398b024">
](url)

[
<img width="521" alt="Screenshot 2023-10-30 235630" src="https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/0f0a7627-513d-490e-99af-d4a3a0e2f0a8">
](url)

[
<img width="562" alt="Screenshot 2023-10-30 235645" src="https://github.com/Dhruv-Jain31/Weather-Notification/assets/126942955/15e0a3cf-9938-48f7-bdd9-d554c7423ff0">
](url)

**Comment by Kashishturan** [comment]
Okay, It was really nice to be in team with you Dhruv. Thanks for being an amazing partner.
